packages/generator/src/artifacts/solid-jsx/jsx.ts

```diff
@@ -36,13 +63,62 @@ export function generateSolidJsxFactory(ctx: Context) {
           const { css: cssStyles, ...propStyles } = styleProps
           const cvaStyles = cvaFn.raw(variantProps)
           return cx(css(cvaStyles, propStyles, cssStyles), localProps.class, localProps.className)
-        }
+        }`
+          })
+          .with('minimal', () => {
+            return outdent`
+            const [localProps, forwardedProps, variantProps, htmlProps, elementProps] = splitProps(
+              mergedProps,
+              ['as', 'class'],
+              forwardedKeys,
+              cvaFn.variantKeys,
+              allCssProperties,
+              normalizeHTMLProps.keys
+            )
+
+            function recipeClass() {
+              const compoundVariantStyles = cvaFn.__getCompoundVariantCss__?.(variantProps);
+              return cx(cvaFn(variantProps, false), css(compoundVariantStyles, combinedProps.css), localProps.class, localProps.className)
+            }
+
+            function cvaClass() {
+              const cvaStyles = cvaFn.raw(variantProps)
+              return cx(css(cvaStyles, combinedProps.css), localProps.class, localProps.className)
+            }`
+          })
+          .with('none', () => {
+            return outdent`
+            const [localProps, forwardedProps, variantProps, htmlProps, elementProps] = splitProps(
+              mergedProps,
+              ['as', 'class'],
+              forwardedKeys,
+              cvaFn.variantKeys,
+              allCssProperties,
+              normalizeHTMLProps.keys
+            )
+
+            function recipeClass() {
+              const compoundVariantStyles = cvaFn.__getCompoundVariantCss__?.(variantProps);
+              return cx(cvaFn(variantProps, false), css(compoundVariantStyles), localProps.class, localProps.className)
+            }
+
+            function cvaClass() {
+              const cvaStyles = cvaFn.raw(variantProps)
+              return cx(css(cvaStyles), localProps.class, localProps.className)
+            }`
+          })
+          .run()}
 
         const classes = configOrCva.__recipe__ ? recipeClass : cvaClass
 
+        if (forwardedProps.className) {
+          delete forwardedProps.className
+        }
```

Comment:
Why are we deleting here?

Reply:
the `splitProps` fn coming from solid doesn't exactly work the same as ours, the `className` ended up in `localProps` AND `forwardedProps` instead of just `localProps`

so when it was later used in the `createComponent(Dynamic, mergeProps(forwardedProps` for some reason that className would have priority vs the `get class () { return classes() }` resulting in not all of the expected classes being attached to the element